src/Microsoft.IdentityModel.Tokens/Json/JsonSerializerPrimitives.cs

```diff
@@ -648,6 +648,33 @@ internal static string ReadString(ref Utf8JsonReader reader, string propertyName
             return reader.GetString();
         }
 
+        /// <summary>
+        /// This method allows a JsonTokenType to be string or number but, it will always return it as a string.
+        /// </summary>
+        /// <param name="reader">The<see cref="Utf8JsonReader"/></param>
+        /// <param name="propertyName">The property name that is being read.</param>
+        /// <param name="className">The type that is being deserialized.</param>
+        /// <param name="read">If true reader.Read() will be called.</param>
+        /// <returns></returns>
+        internal static string ReadNumberAsString(ref Utf8JsonReader reader, string propertyName, string className, bool read = false)
```

Comment:
This name isn't honest. Reads as either string or number isn't ReadNumberAsString

Reply:
it is not public api, so no strong opinions. As long as it is honest 😄

Reply:
ReadSringOrNumberAsString it is lol